### ISSUE TITLE
PlaybackStatsListener causes crash

### ISSUE THREAD

**Issue by indieboyalex** [google/ExoPlayer#8772]
ExoPlayer 2.13.2

Upgrading from ExoPlayer 2.12.3 -> 2.13.2 I'm finding that everything works fine, but when I close my PlayerActivity my app is crashing.  The error we get is:

```
java.lang.IllegalArgumentException
        at com.google.android.exoplayer2.util.Assertions.checkArgument(Assertions.java:41)
        at com.google.android.exoplayer2.analytics.PlaybackStatsListener$PlaybackStatsTracker.updatePlaybackState(PlaybackStatsListener.java:679)
        at com.google.android.exoplayer2.analytics.PlaybackStatsListener$PlaybackStatsTracker.onFinished(PlaybackStatsListener.java:477)
        at com.google.android.exoplayer2.analytics.PlaybackStatsListener.onSessionFinished(PlaybackStatsListener.java:183)
        at com.google.android.exoplayer2.analytics.DefaultPlaybackSessionManager.finishAllSessions(DefaultPlaybackSessionManager.java:247)
        at com.google.android.exoplayer2.analytics.PlaybackStatsListener.onEvents(PlaybackStatsListener.java:291)
        at com.google.android.exoplayer2.analytics.AnalyticsCollector.lambda$setPlayer$1$AnalyticsCollector(AnalyticsCollector.java:141)
        at com.google.android.exoplayer2.analytics.-$$Lambda$AnalyticsCollector$yS3z67Oa1f_t9PmuRvwn2pyVDcQ.invoke(Unknown Source:8)
        at com.google.android.exoplayer2.util.ListenerSet$ListenerHolder.release(ListenerSet.java:287)
        at com.google.android.exoplayer2.util.ListenerSet.release(ListenerSet.java:230)
        at com.google.android.exoplayer2.util.ListenerSet.handleMessage(ListenerSet.java:266)
        at com.google.android.exoplayer2.util.ListenerSet.lambda$eEvjP-IE0x3J2lRvKfFbbjRFRvc(Unknown Source:0)
        at com.google.android.exoplayer2.util.-$$Lambda$ListenerSet$eEvjP-IE0x3J2lRvKfFbbjRFRvc.handleMessage(Unknown Source:2)
        at android.os.Handler.dispatchMessage(Handler.java:108)
        at android.os.Looper.loop(Looper.java:216)
        at android.app.ActivityThread.main(ActivityThread.java:7625)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
```

I've noticed that in our code:
```
playerStatsListener = PlaybackStatsListener(false, null)
addAnalyticsListener(playerStatsListener!!)
```
if I simply comment out the addAnalyticsListener call then the crash does not happen.

Is this crash a new problem with PlaybackStatsListener, and is there a fix for it?

A secondary question would be, given that we're not using a callback with PSL, is there any drawback to simple not adding the listener (at least until such time as the crash is fixed)?

I'm not sure if this counts as a separate issue to https://github.com/google/ExoPlayer/issues/8675 or not as we are using DRM protected Widevine-DASH VOD content, not a live stream.

**Comment by tonihei** [comment]
Looks like the same issue as #8675, which may occur if you reset the same player with new media or use another player with the same listener attached, AND a seek happens just before the new playback starts. You said it's happening when you close your Activity? Can you check whether something in your code is attempting to seek just before you release the player?

**Comment by indieboyalex** [comment]
I can't see anything in the code that would be causing a seek and none of the available logging suggests any kind of seek is going on.

**Comment by tonihei** [comment]
If you have a way to reproduce, can you describe what we'd need to do to see the same problem here or upload a small example project showing the problem? 

Alternatively, it's probably quicker to wait until we uploaded the fix for #8675 and test against the ExoPlayer version with this fix to see if it's indeed the same problem.

**Comment by indieboyalex** [comment]
It looks like if I remove the PSL from the player before calling release then I don't get the crash (at least this seems to be working from the limited testing I've done so far)

**Comment by tonihei** [comment]
Yes, this is avoiding the issue, but I'm trying to find out whether there is something we need to fix here besides the fix for #8675.

**Comment by indieboyalex** [comment]
Well from what I've looked at today, what I've seen is:
* There was no issue in Exo 2.12, but only once we upgraded - so it's a new behaviour
* I could not find any `seek` being called anywhere.  This was happening in both player versions in our app (1 for mobile, 1 for TV).  The player(s) are however very complicated so I'd probably only be able to be 90% confident this wasn't the issue. Apologies that I can't be 100% confident of this.
* Workaround mentioned above does at least work (so we can complete the version update for our app at any rate)

**Comment by VivekBhalodiyaMM** [comment]
I'm having the same issue but with a different scenario. 
To reproduce
1. Start the playback
2. Call `playbackStatsListener.finishAllSessions()` in  the onTaskRemoved of your service
3. Followed by `exoplayer.stop()`
4. Kill and relaunch the app, play the content, which will call `exoPlayer.prepare(mediaSource)` which causes the crash.

Note - There is also seek happening after `explyer.prepare(mediaSource)`
ExoPlayer version - 2.12

**Comment by tonihei** [comment]
@VivekBhalodiyaMM  This is unrelated and old bug that got fixed in 2.13.0

**Comment by VivekBhalodiyaMM** [comment]
> @VivekBhalodiyaMM This is unrelated and old bug that got fixed in 2.13.0

Thank you @tonihei I'll verify and check.

**Comment by indieboyalex** [comment]
As an update to this:

We attempted to do as much as we could to ensure the analytics listeners were removed before releasing the player.  This resolved any crashing issues we were seeing locally.  However out in the wild we were still seeing a huge volume of crashes related to this Assertion check, and nothing seemed to suggest a problem on our side.  Eventual solution was to clone the repo, and compile with assertions disabled.  Bingo - no more crashes.  It was either this or revert to the previously stable version of ExoPlayer before the crashes started happening.
